### ISSUE TITLE
ASM module erroring out on a count var

### ISSUE THREAD

**Issue by pramodsetlur** [terraform-google-modules/terraform-google-kubernetes-engine#1259]
### TL;DR

Error: Invalid count argument

  on .terraform/modules/gke_cluster.asm_managed_control_plane.cpr/main.tf line 57, in resource "random_id" "cache":
  57:   count = (!local.skip_download) ? 1 : 0

The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.

### Terraform Configuration

```hcl
# Managed ASM inside GKE
module "asm_managed_control_plane" {
  depends_on = [
    google_container_cluster.gke_cluster
  ]
  count = var.asm_managed_control_plane_enabled == true ? 1 : 0
  source                    = "github.com/terraform-google-modules/terraform-google-kubernetes-engine//modules/asm?ref=v21.0.0"
  project_id                = var.project_id
  cluster_name              = var.cluster_name
  cluster_location          = var.region
  enable_cni                = true
  enable_fleet_registration = true
  enable_mesh_feature       = false
}

The above depends on the below GKE cluster resource.
resource "google_container_cluster" "gke_cluster" {
  provider = google-beta

  name     = var.cluster_name
  location = var.region
  .
  .
  .
}
```

### Terraform Version

```sh
Terraform version: "0.14.8"
```

### Additional information

I was initially pointing to master and then updated it to v21.0.0 after #1189 was merged and released. It had been working last week when I created a new cluster with managed ASM control plane, but facing this error now.
